polyfill/lib/zoneddatetime.mjs

```diff
@@ -0,0 +1,460 @@
+    }
+    options = ES.NormalizeOptionsObject(options);
+    const smallestUnit = ES.ToSmallestTemporalDurationUnit(options, 'nanoseconds');
+    const defaultLargestUnit = ES.LargerOfTwoTemporalDurationUnits('days', smallestUnit);
+    const largestUnit = ES.ToLargestTemporalUnit(options, defaultLargestUnit);
+    ES.ValidateTemporalUnitRange(largestUnit, smallestUnit);
+    let roundingMode = ES.ToTemporalRoundingMode(options, 'nearest');
+    roundingMode = ES.NegateTemporalRoundingMode(roundingMode);
+    const maximumIncrements = {
+      years: undefined,
+      months: undefined,
+      weeks: undefined,
+      days: undefined,
+      hours: 24,
+      minutes: 60,
+      seconds: 60,
+      milliseconds: 1000,
+      microseconds: 1000,
+      nanoseconds: 1000
+    };
+    const roundingIncrement = ES.ToTemporalRoundingIncrement(options, maximumIncrements[smallestUnit], false);
+    void roundingMode;
+    void roundingIncrement;
+    throw new Error('since() not implemented yet');
+  }
+  round(options) {
+    if (!ES.IsTemporalZonedDateTime(this)) throw new TypeError('invalid receiver');
+    if (options === undefined) throw new TypeError('options parameter is required');
+    options = ES.NormalizeOptionsObject(options);
+    const smallestUnit = ES.ToSmallestTemporalUnit(options);
+    const roundingMode = ES.ToTemporalRoundingMode(options, 'nearest');
+    const maximumIncrements = {
+      day: 1,
+      hour: 24,
+      minute: 60,
+      second: 60,
+      millisecond: 1000,
+      microsecond: 1000,
+      nanosecond: 1000
+    };
+    const roundingIncrement = ES.ToTemporalRoundingIncrement(options, maximumIncrements[smallestUnit], false);
+    void roundingMode;
+    void roundingIncrement;
+    throw new Error('round() not implemented yet');
+  }
+  equals(other) {
+    if (!ES.IsTemporalZonedDateTime(this)) throw new TypeError('invalid receiver');
+    other = ES.ToTemporalZonedDateTime(other, ZonedDateTime);
+    const one = GetSlot(this, EPOCHNANOSECONDS);
+    const two = GetSlot(other, EPOCHNANOSECONDS);
+    if (!bigInt(one).equals(two)) return false;
+    if (!ES.TimeZoneEquals(GetSlot(this, TIME_ZONE), GetSlot(other, TIME_ZONE))) return false;
+    return ES.CalendarEquals(GetSlot(this, CALENDAR), GetSlot(other, CALENDAR));
+  }
+  toString(options = undefined) {
+    if (!ES.IsTemporalZonedDateTime(this)) throw new TypeError('invalid receiver');
+    options = ES.NormalizeOptionsObject(options);
+    const { precision, unit, increment } = ES.ToSecondsStringPrecision(options);
+    const roundingMode = ES.ToTemporalRoundingMode(options, 'trunc');
+    return zonedDateTimeToString(this, precision, { unit, increment, roundingMode });
+  }
+  toLocaleString(locales = undefined, options = undefined) {
+    if (!ES.IsTemporalZonedDateTime(this)) throw new TypeError('invalid receiver');
+    return new DateTimeFormat(locales, options).format(this);
```

Comment:
All the toLocaleString logic happens over in Intl? I'm particularly wondering about the case where the time zone of the receiver conflicts with the time zone in the options. #700 forces the instance's time zone into the options bag and throws an exception if the user put a timezone in the options that's different from the instance's time zone. I think (not 100% sure) matches how calendar conflicts are handled which seems like a reasonable behavior to match.

Reply:
Yeah, the check is in [PartitionDateTimePattern](https://tc39.es/proposal-temporal/#sec-partitiondatetimepattern); we wouldn't want `toLocaleString` to be inconsistent with manually using `DateTimeFormat`.